Runtime/src/test/java/com/rakuten/tech/mobile/perf/runtime/internal/ConfigurationRequestSpec.java

```diff
@@ -33,27 +33,88 @@
   }
 
   @Test public void shouldBuildUrlWithDefaultUrlPrefix() {
-    ConfigurationRequest request = new ConfigurationRequest(null, "", builder.build(), null, null);
+    ConfigurationRequest request = new ConfigurationRequest(
+        null, "", builder.build(), null, null);
     assertThat(request.getUrl())
-        .isEqualTo(BuildConfig.DEFAULT_CONFIG_URL_PREFIX +
-            "/platform/testPlatform/app/testId/version/testVersion/?sdk=testSdkVersion&country=testCountryCode");
+        .startsWith(BuildConfig.DEFAULT_CONFIG_URL_PREFIX);
   }
 
   @Test public void shouldBuildUrlWithCustomDomain() {
     ConfigurationRequest request = new ConfigurationRequest("https://rakuten.co.jp", "", builder
         .build(), null, null);
     assertThat(request.getUrl())
-        .isEqualTo(
-            "https://rakuten.co.jp/platform/testPlatform/app/testId/version/testVersion/?sdk=testSdkVersion&country=testCountryCode");
+        .startsWith("https://rakuten.co.jp/");
   }
 
   @Test public void shouldBuildUrlWithCustomPrefix() {
     ConfigurationRequest request =
         new ConfigurationRequest("https://other.prefix.com/abc/xyz/v1",
             "", builder.build(), null, null);
     assertThat(request.getUrl())
-        .isEqualTo(
-            "https://other.prefix.com/abc/xyz/v1/platform/testPlatform/app/testId/version/testVersion/?sdk=testSdkVersion&country=testCountryCode");
+        .startsWith("https://other.prefix.com/abc/xyz/v1/");
+  }
+
+  @Test public void shouldBuildUrlWithPlatform() {
+    builder.setPlatform("testPlatform");
+
+    ConfigurationRequest request = new ConfigurationRequest(
+        null, "", builder.build(), null, null);
+    assertThat(request.getUrl())
+        .contains("/platform/testPlatform/");
+  }
+
+  @Test public void shouldBuildUrlWithAppId() {
+    builder.setAppId("testId");
+
+    ConfigurationRequest request =
+        new ConfigurationRequest(null, "", builder.build(), null, null);
+    assertThat(request.getUrl())
+        .contains("/app/testId/");
+  }
+
+  @Test public void shouldBuildUrlWithAppVersion() {
+    builder.setAppVersion("testVersion");
+
+    ConfigurationRequest request = new ConfigurationRequest(
+        null, "", builder.build(), null, null);
+    assertThat(request.getUrl())
+        .contains("/version/testVersion/");
+  }
+
+  @Test public void shouldBuildUrlWithSdkVersion() {
+    builder.setSdkVersion("testSdkVersion");
+
+    ConfigurationRequest request = new ConfigurationRequest(
+        null, "", builder.build(), null, null);
+    assertThat(request.getUrl())
+        .contains("?sdk=testSdkVersion");
+  }
+
+  @Test public void shouldBuildUrlWithCountryCode() {
+    builder.setCountryCode("testCountryCode");
+
+    ConfigurationRequest request = new ConfigurationRequest(
+        null, "", builder.build(), null, null);
+    assertThat(request.getUrl())
+        .contains("&country=testCountryCode");
```

Comment:
the fact that `country` is the first parameter is an implementation detail, isn't it?